Chiazokam/XamBox
# #162933343 create signin endpoint

#### What does this PR do?
- creates the user sign in controller
- creates the signin route and include middlewares
- writes tests to cover all the functions

#### Description of Task to be completed?
- create the user sign in controller
- create the signin route and include middlewares
- write tests to cover all the functions

#### How should this be manually tested?
- Clone the repo - `https://github.com/Chiazokam/xambox`
- Enter into the project directory - `cd xambox`
- switch to this branch - `checkout ft-api-signin-162933343`
- Start the server `npm run start:dev`
- Open `Postman`
- On `postman` Send a `POST` request to `http://localhost:3000/api/v1/signin`

- You should receive a successful response

#### Any background context you want to provide?
Input data of the format:
`{
                "email": "myfrankly@frank.com",
                "password": "mayorfranklyn"
              }`

#### What are the relevant pivotal tracker stories?
[#162933343](https://www.pivotaltracker.com/story/show/162933343)
